fpga_interchange/dedicated_interconnect.cc

```diff
@@ -0,0 +1,785 @@
+                const BelInfoPOD &src_bel_info = src_tile_type.bel_data[src.bel_index];
+                IdString src_site_type = IdString(src_tile_type.site_types[src_bel_info.site]);
+                IdString src_bel_pin = src.bel_pin;
+
+                const TileTypeInfoPOD &dst_tile_type = ctx->chip_info->tile_types[dst.tile_type];
+                const BelInfoPOD &dst_bel_info = dst_tile_type.bel_data[dst.bel_index];
+                IdString dst_site_type = IdString(dst_tile_type.site_types[dst_bel_info.site]);
+                IdString dst_bel_pin = dst.bel_pin;
+
+                log_info("%s.%s[%d]/%s/%s (%d, %d) -> %s.%s[%d]/%s/%s\n", IdString(src_tile_type.name).c_str(ctx),
+                         src_site_type.c_str(ctx), src_bel_info.site, IdString(src_bel_info.name).c_str(ctx),
+                         src_bel_pin.c_str(ctx), delta_x, delta_y, IdString(dst_tile_type.name).c_str(ctx),
+                         dst_site_type.c_str(ctx), dst_bel_info.site, IdString(dst_bel_info.name).c_str(ctx),
+                         dst_bel_pin.c_str(ctx));
+            }
+        } else {
+            auto src = key;
+            for (const auto &dst_delta : sources.at(key)) {
+                auto dst = dst_delta.type_bel_pin;
+                auto delta_x = dst_delta.delta_x;
+                auto delta_y = dst_delta.delta_y;
+
+                const TileTypeInfoPOD &src_tile_type = ctx->chip_info->tile_types[src.tile_type];
+                const BelInfoPOD &src_bel_info = src_tile_type.bel_data[src.bel_index];
+                IdString src_site_type = IdString(src_tile_type.site_types[src_bel_info.site]);
+                IdString src_bel_pin = src.bel_pin;
+
+                const TileTypeInfoPOD &dst_tile_type = ctx->chip_info->tile_types[dst.tile_type];
+                const BelInfoPOD &dst_bel_info = dst_tile_type.bel_data[dst.bel_index];
+                IdString dst_site_type = IdString(dst_tile_type.site_types[dst_bel_info.site]);
+                IdString dst_bel_pin = dst.bel_pin;
+
+                log_info("%s.%s[%d]/%s/%s -> %s.%s[%d]/%s/%s  (%d, %d)\n", IdString(src_tile_type.name).c_str(ctx),
+                         src_site_type.c_str(ctx), src_bel_info.site, IdString(src_bel_info.name).c_str(ctx),
+                         src_bel_pin.c_str(ctx), IdString(dst_tile_type.name).c_str(ctx), dst_site_type.c_str(ctx),
+                         dst_bel_info.site, IdString(dst_bel_info.name).c_str(ctx), dst_bel_pin.c_str(ctx), delta_x,
+                         delta_y);
+            }
+        }
+    }
+}
+
+void DedicatedInterconnect::find_dedicated_interconnect()
+{
+    for (BelId bel : ctx->getBels()) {
+        const auto &bel_data = bel_info(ctx->chip_info, bel);
+        if (bel_data.category != BEL_CATEGORY_LOGIC) {
+            continue;
+        }
+        if (bel_data.synthetic) {
+            continue;
+        }
+
+        for (size_t i = 0; i < bel_data.num_bel_wires; ++i) {
```

Comment:
mixed signedness compare warning (I'm minded to tighten up the compiler flags so more of these things become CI failures, not sure what your opinion is on that)

Reply:
Please tighten up the flags.  We also should emit `-Wxxx` on local builds.  I'm using `clang` and it is capable of emitting the same warnings as the CI, but that isn't the current default (I believe).